skpro/distributions/concat.py

```diff
@@ -0,0 +1,96 @@
+# copyright: skpro developers, BSD-3-Clause License (see LICENSE file)
+"""Concat operation."""
+
+__author__ = ["SaiRevanth25"]
+
+import pandas as pd
+
+
+class concat:
```

Comment:
I think we need two things:

* a `function` `concat`, with signature like `pandas.concat`. It produces a `ConcatDistr` when passed a list of distr.
* a class `ConcatDistr`, that inherits `_HeterogenousMetaObject`
    * this looks like the current class

Although, now that I am thinking about this, it might be worth about just calling the class `concat`. Although, perhaps it is best not to do that, since it does not align with the `UpperCamelCase` naming convention.